cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java

```diff
@@ -68,7 +79,12 @@ public void encrypt() {
     protected String encrypt(Serializable originalContent) {
         byte[] serialized = serialize(originalContent);
         String serializedAsString = Base64.encodeBase64String(serialized);
-        return EncryptionProvider.get().encrypt(serializedAsString.toCharArray());
+        Encryption encryption = EncryptionProvider.get();
```

Comment:
Nice catch @Bonczidai .
I think indeed the encryptor deobfuscates and decrypts, but I think this is an implementation side effect and we should provide the spi in the best possible way using `deobfuscate` method in the contract with symmetric logic for both encryption and decryption.